twilight-model/src/gateway/event/mod.rs

```diff
@@ -18,8 +18,9 @@ use std::fmt::{Debug, Display, Formatter, Result as FmtResult};
 
 /// Any type of event that a shard emits.
 ///
-/// This brings together all of the types of [`DispatchEvent`] and
-/// [`GatewayEvent`].
+/// Flattened type with all of the variants of [`DispatchEvent`] and
+/// [`GatewayEvent`] together with gateway close messages and their optional
+/// [`CloseFrame`].
```

Comment:
This feels... off, and I can't put my finger on why. It may be because saying "gateway close messages and their close frames" feels oddly specific about what close messages contain, while not specifying what `DispatchEvent` and `GatewayEvent` contain. Maybe we could go with removing the `CloseFrame` mention and linking to the gateway close variant? It feels like it bridges the gap of the gateway close variant not being linked while also allowing users to quickly link into it and learn that it *does* contain a `CloseFrame`.
```suggestion
/// Flattened type containing all the variants of [`DispatchEvent`] and
/// [`GatewayEvent`], along with a [gateway close event].
///
/// [gateway close event]: Self::GatewayClose
```